pds/contracts/PackNFT.cdc

```diff
@@ -88,168 +110,303 @@ pub contract PackNFT: NonFungibleToken, IPackNFT {
-        // and adds the ID to the id array
-        pub fun deposit(token: @NonFungibleToken.NFT) {
-            let token <- token as! @PackNFT.NFT
-
+        /// Deposit an NFT into this Collection.
+        ///
+        access(all) fun deposit(token: @{NonFungibleToken.NFT}) {
+            let token <- token as! @NFT
             let id: UInt64 = token.id
-
-            // add the new token to the dictionary which removes the old one
+            // Add the new token to the dictionary which removes the old one.
             let oldToken <- self.ownedNFTs[id] <- token
-            emit Deposit(id: id, to: self.owner?.address)
 
+            // Deposited event emitted from NonFungibleToken contract interface.
+            emit Deposit(id: id, to: self.owner?.address) // TODO: Consider removing
             destroy oldToken
         }
 
-        // getIDs returns an array of the IDs that are in the collection
-        pub fun getIDs(): [UInt64] {
+        /// Return an array of the IDs that are in the collection.
+        ///
+        access(all) view fun getIDs(): [UInt64] {
             return self.ownedNFTs.keys
         }
 
-        // borrowNFT gets a reference to an NFT in the collection
-        // so that the caller can read its metadata and call its methods
-        pub fun borrowNFT(id: UInt64): &NonFungibleToken.NFT {
-            return (&self.ownedNFTs[id] as &NonFungibleToken.NFT?)!
+        /// Return the amount of NFTs stored in the collection.
+        ///
+        access(all) view fun getLength(): Int {
+            return self.ownedNFTs.keys.length
+        }
+
+        /// Return a list of NFT types that this receiver accepts.
+        ///
+        access(all) view fun getSupportedNFTTypes(): {Type: Bool} {
+            let supportedTypes: {Type: Bool} = {}
+            supportedTypes[Type<@NFT>()] = true
+            return supportedTypes
+        }
+
+        /// Return whether or not the given type is accepted by the collection.
+        ///
+        access(all) view fun isSupportedNFTType(type: Type): Bool {
+            if type == Type<@NFT>() {
+                return true
+            }
+            return false
+        }
+
+        /// Return a reference to an NFT in the Collection.
+        ///
+        access(all) view fun borrowNFT(_ id: UInt64): &{NonFungibleToken.NFT}? {
+            return &self.ownedNFTs[id]
         }
 
-        pub fun borrowPackNFT(id: UInt64): &IPackNFT.NFT? {
-            let nft <- self.ownedNFTs.remove(key: id) ?? panic("missing NFT")
-            let token <- nft as! @PackNFT.NFT
-            let ref = &token as &IPackNFT.NFT
-            self.ownedNFTs[id] <-! token as! @PackNFT.NFT
-            return ref
+        /// Return a reference to an NFT in the Collection as a IPackNFT.NFT.
+        ///
+        access(all) view fun borrowPackNFT(id: UInt64): &{IPackNFT.NFT}? {
+            return self.borrowNFT(id) as! &{IPackNFT.NFT}?
         }
```

Comment:
This function is not needed any more since we have borrowNFT now with safe downcasting. You don't need to remove it, but it could clean up the code a bit

Reply:
Cleaned up - thanks!